### PR TITLE
EventGrid: Updated to use the latest version of the .NET SDK

### PR DESCRIPTION
**Pull request by kalyanaj**
## Description
EventGrid: Updated to use the latest version of the .NET SDK and re-recorded the tests.

---

This checklist is used to make sure that common guidelines for a pull request are followed. You can find a more complete discussion of PowerShell cmdlet best practices [here](https://msdn.microsoft.com/en-us/library/dd878270(v=vs.85).aspx).

- [X] **I have read the [contribution guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md).**
- [X] **If changes were made to any cmdlet, the XML help was regenerated using the [platyPSHelp module](https://github.com/Azure/azure-powershell/blob/preview/documentation/help-generation.md).**
- [X] **If any large changes are made to a service, they are reflected in the respective [change log](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#updating-the-change-log).**

### [General Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#general-guidelines)
- [X] Title of the pull request is clear and informative.
- [X] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/preview/documentation/cleaning-up-commits.md).
- [X] The pull request does not introduce [breaking changes](https://github.com/Azure/azure-powershell/blob/preview/documentation/breaking-changes/breaking-changes-definition.md) (unless a major version change occurs in the assembly and module).

### [Testing Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#testing-guidelines)
- [X] Pull request includes test coverage for the included changes.
- [X] PowerShell scripts used in tests should do any necessary setup as part of the test or suite setup, and should not use hard-coded values for locations or existing resources.

### [Cmdlet Signature Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-signature-guidelines)
- [X] New cmdlets that make changes or have side effects should implement `ShouldProcess` and have `SupportShouldProcess=true` specified in the cmdlet attribute. You can find more information on `ShouldProcess` [here](https://github.com/Azure/azure-powershell/wiki/PowerShell-Cmdlet-Design-Guidelines#supportsshouldprocess).
- [X] Cmdlet specifies `OutputType` attribute if any output is produced - if the cmdlet produces no output, it should implement a `PassThru` parameter.

### [Cmdlet Parameter Guidelines](https://github.com/Azure/azure-powershell/blob/preview/CONTRIBUTING.md#cmdlet-parameter-guidelines)
- [X] Parameter types should not expose types from the management library - complex parameter types should be defined in the module.
- [X] Complex parameter types are discouraged - a parameter type should be simple types as often as possible. If complex types are used, they should be shallow and easily creatable from a constructor or another cmdlet.
- [X] Cmdlet parameter sets should be mutually exclusive - each parameter set must have at least one mandatory parameter not in other parameter sets.

### CODE REVIEW

**Comment by cormacpayne** [comment]
@kalyanaj you can revert the version changes that you are making - we will update the versions for you during the release, or if you would like to release this module out-of-band, then we will use our (under construction) versioning tool to determine the proper version bump for the module.

File: src/ResourceManager/EventGrid/Commands.EventGrid.Test/ScenarioTests/EventSubscriptionTests.ps1 [hunk 1]
```diff
@@ -153,15 +153,18 @@ function EventSubscriptionTests_ResourceGroup {
...
-	$updatedLabels = $updateResult.Labels
-	Assert-AreEqual 2 $updatedLabels.Count;
-    Assert-AreEqual "Marketing" $updatedLabels[0];
-    Assert-AreEqual "Sales" $updatedLabels[1];
+    # Currently blocked by a EventGrid Service issue, hence commenting it out for now until the Service fix gets deployed.
+    # TODO: Uncomment this once the Service fix gets deployed.
+
+    # Write-Debug "Updating eventSubscription $eventSubscriptionName to resource group $resourceGroupName"
+    # $newLabels = "Marketing", "Sales"
+    # $updateResult = Update-AzureRmEventGridSubscription -ResourceGroup $resourceGroupName -EventSubscriptionName $eventSubscriptionName -SubjectEndsWith "NewSuffix" -Label $newLabels
```
[the remark this answers is not in the document]
Comment (kalyanaj) [reply src/ResourceManager/EventGrid/Commands.EventGrid.Test/ScenarioTests/EventSubscriptionTests.ps1]: It impacted all API versions. A service fix was deployed yesterday that resolves this issue - I have uncommented out these lines and am re-recording the tests to re-include this change.

File: src/ResourceManager/EventGrid/Commands.EventGrid/Utilities/EventGridClient.cs [hunk 2]
```diff
@@ -178,7 +178,7 @@ public TopicSharedAccessKeys RegenerateTopicKey(string resourceGroupName, string
                 eventSubscription.Labels = new List<string>(labels);
             }
 
-            return this.Client.EventSubscriptions.Create(scope, eventSubscriptionName, eventSubscription);
+            return this.Client.EventSubscriptions.CreateOrUpdate(scope, eventSubscriptionName, eventSubscription);
```
[the remark this answers is not in the document]
Comment (kalyanaj) [reply src/ResourceManager/EventGrid/Commands.EventGrid/Utilities/EventGridClient.cs]: No, it shouldn't. Update-AzureRmEventGridSubscription maps to a PATCH request and can be used to perform a partial update of the resource (e.g. in the case of event subscriptions, I want to update only the "filter" and not the "destination"). New-AzureRmEventGridSubscription maps to a PUT request which either creates or replaces the resource.

**Comment by cormacpayne** [comment]
On-demand: https://azuresdkci.westus2.cloudapp.azure.com/job/powershell-demand/394/

**Pull request merged**